CookSmart/CookSmart/UI/IngredientListView.swift

```diff
@@ -0,0 +1,263 @@
+//
+//  IngredientListView.swift
+//  cake
+//
+//  Created by Vova Galchenko on 11/25/23.
+//  Copyright © 2023 Olga Galchenko. All rights reserved.
+//
+
+import SwiftUI
+
+struct IngredientListView: View {
+  @StateObject private var ingredientsStore: IngredientsStore
+  @State private var searchText: String = ""
+  @State private var isShowingResetAlert = false
+  private var ingredientGroupsToPresent: [any IngredientGroup] {
+    let relevantGroups = searchText.isEmpty ?
+      ingredientsStore.ingredientGroups :
+      ingredientsStore.ingredientGroups.compactMap { group in
+        group.filter(searchString: searchText)
+      }
+    return relevantGroups.filter { !$0.ingredients.isEmpty }
+  }
+
+  weak var delegate: IngredientListViewDelegate?
+
+  init(
+    ingredientsStore: @escaping @autoclosure () -> IngredientsStore = IngredientsStore.shared,
+    delegate: IngredientListViewDelegate? = nil
+  ) {
+    _ingredientsStore = StateObject(wrappedValue: ingredientsStore())
+    self.delegate = delegate
+  }
+
+  var body: some View {
+    NavigationView {
+      List {
+        ForEach(ingredientGroupsToPresent, id: \.id) { group in
+          // I actually think in this case it's more readable to keep this code inline,
+          // than to extract it into IngredientGroupSection, because of the strong ties
+          // it has back to this struct, but the compiler crashes when the section code
+          // is here inline :(
+          IngredientGroupSection(group: group, delegate: delegate)
+        }
+      }
+      .listStyle(PlainListStyle())
+      // O&A Question: There's apparently no way to customize the appearance of the search bar... what do you suggest?
```

Comment:
Hm I've never used ".searchable". We could just add our own search bar I guess. Are we just trying to style it like the rest of the app?